modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java

```diff
@@ -256,6 +320,71 @@ public void testGetMetaOutboundConnector() throws IOException {
                 .body("rulesEnabled", equalTo(false));
     }
 
+    @Test
+    public void testAddIdPWithUserDefinedAuthenticator() throws IOException {
+
+        String body = idpCreatePayload.replace(FEDERATED_AUTHENTICATOR_ID_PLACEHOLDER,
+                userDefinedAuthenticatorPayload.getAuthenticatorId());
+        body = body.replace(FEDERATED_AUTHENTICATOR_PLACEHOLDER,
+                userDefinedAuthenticatorPayload.convertToJasonPayload());
+        body = body.replace(IDP_NAME_PLACEHOLDER, IDP_NAME);
+        Response response = getResponseOfPost(IDP_API_BASE_PATH, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+        customIdPId = location.substring(location.lastIndexOf("/") + 1);
+        assertNotNull(customIdPId);
+    }
+
+    @Test(dependsOnMethods = "testAddIdPWithUserDefinedAuthenticator")
+    public void testGetUserDefinedAuthenticatorsOfIdP() {
+
+        Response response = getResponseOfGet(IDP_API_BASE_PATH + PATH_SEPARATOR + customIdPId +
```

Comment:
Is the customIdPId is updated? Make sure all endpoint configuration updates are covered.

Reply:
This is updated with 38608298188e4987329c170552b4a0de9fb72daf